src/lib/models/workflow-execution.ts

```diff
@@ -39,41 +21,54 @@ export interface WorkflowExecution {
   taskQueue?: string;
   historyEvents: Long;
   pendingActivities: PendingActivityInfo[];
-}
+  url: string;
+  toggleUrl: (isFullScreen: boolean) => string;
+};
 
 type WorkflowExecutionAPIResponse = Optional<
   DescribeWorkflowExecutionResponse,
   'executionConfig' | 'pendingActivities' | 'pendingChildren'
 >;
 
-export class WorkflowExecution {
-  constructor({
-    executionConfig,
-    workflowExecutionInfo,
-    pendingActivities,
-  }: WorkflowExecutionAPIResponse) {
-    this.name = workflowExecutionInfo.type.name;
-    this.id = workflowExecutionInfo.execution.workflowId;
-    this.runId = workflowExecutionInfo.execution.runId;
-    this.startTime = formatDate(workflowExecutionInfo.startTime);
-    this.endTime = formatDate(workflowExecutionInfo.closeTime);
-    this.status = workflowExecutionInfo.status;
-    this.historyEvents = workflowExecutionInfo.historyLength;
-
-    if (executionConfig) {
-      this.taskQueue = executionConfig.taskQueue.name;
-    }
-
-    this.pendingActivities = pendingActivities;
-  }
-
-  get url(): string {
-    return `/workflows/${this.id}/${this.runId}`;
-  }
+export const toWorkflowExecution = (
```

Comment:
Instead of having a function that invokes a class, we're just creating a plain ol' JavaScript object with everything it needs.